joomla/joomla-cms
# [3.9][Privacy] Block usage of FLoC by default

### Issue
As a replacement for third party cookies, Google has introduced the Federated Learning of Cohorts into the Chrome browser lately. This feature is supposed to allow better tracking of users while also keeping data privacy. Details can be read [here](https://github.com/WICG/floc). The Joomla! project disagrees that this feature is in the interests of the owners of Joomla!-powered websites as well as their visitors. We share the [assessment](https://www.eff.org/deeplinks/2021/03/googles-floc-terrible-idea) of the EFF as well as many other organisations ([Mozilla, Microsoft, Opera](https://www.theverge.com/2021/4/16/22387492/google-floc-ad-tech-privacy-browsers-brave-vivaldi-edge-mozilla-chrome-safari), [WordPress]( https://make.wordpress.org/core/2021/04/18/proposal-treat-floc-as-a-security-concern/)).

We consider this a security issue and thus will roll out countermeasures with the next bugfix release of the Joomla 3.9 series.

### Summary of Changes
This PR introduces a header to disable Federated Learning of Cohorts, which is sent with every request that is handled by the Joomla framework. The header specifically looks like this:
`
Permissions-Policy: interest-cohort=()
`
and is the equivalent of opting out of this feature. Please be aware that this only adds this header to requests which go through the Joomla! application. Assets like CSS files, images or javascript are handled by your webserver directly and we would advise to modify the webserver to add this header to every request directly.

If you really want to disable the blocking of this feature, we added a switch in the global configuration to remove this header.

A postinstall message has been added to the database to inform administrators of this decision.

### Testing Instructions
Apply this patch and load a random page of your testing site. The answer to your request should contain the above noted header.

### Documentation Changes Required
None.

### Handling of this in the future
This PR fixes this issue for the 3.9 and 3.10 releases. We will decide on how to implement this countermeasure in 4.0 in the time before the final release.
